CIP-XXXX/x509-roles.cddl

```diff
@@ -3,34 +3,42 @@
 ; Cardano ledger implementation.
 
 ; uses 
-x509_roles = $x509_roles .within role-structure
+x509-roles = $x509-roles .within role-structure
 
 role-structure = [ role-version, role-body ]
 
 role-version = uint ; Version is an integer.
 role-body = { + uint => any } ; General role body is a Map of integers to anything.
 
 ; Version 0 of the role registration format.
-$x509_roles /= x509-roles-v0
+$x509-roles /= x509-roles-v0
 
 x509-roles-v0 = [ 0, x509-roles-v0-body ]
 
 x509-roles-v0-body = {
-	? x509-certs => [ + x509_der_cert ], ; Un-ordered List of DER encoded x509 certificates
-	? c509-certs => [ + c509_cert ], ; Un-ordered List of CBOR encoded C509 certificates (or metadatum references)
-	? pub-keys => [ + simple-public-key-type ], ; Ordered list of simple public keys that are registered.
-	? revocation-list => [ + revocation-entry ], ; Revocation list of certs being revoked by an issuer.
+	? x509-certs => [ * x509-der-cert ], ; Ordered List of DER encoded x509 certificates
```

Comment:
These should all be `+`.  `*` means zero or more.  `+` means 1 or more.  IF there are no updates to a cert or revocation list, the list should not be produced, so it should only be produced if it has at least 1 element in all cases.
Otherwise a list like [] is valid, but its not necessary, so we should restrict it.